src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -980,24 +1026,129 @@ private void doEncodeTransfer(
+                .size(size)
+                .doff(2)
+                .type(0)
+                .channel(channel)
+                .performative(b -> b.transfer(transfer))
+                .build();
+            payloadRW.put(frameHeader.buffer(), frameHeader.offset(), frameHeader.sizeof());
+            return frameHeader;
+        }
+
+        private AmqpMapFW<AmqpValueFW, AmqpValueFW> annotations(
+            ArrayFW<AmqpAnnotationFW> annotations,
+            int bufferOffset)
+        {
+            AmqpMapFW<AmqpValueFW, AmqpValueFW> messageAnnotations = null;
+            if (annotations.fieldCount() > 0)
+            {
+                amqpMapRW.wrap(extraBuffer, bufferOffset, extraBuffer.capacity());
+                annotations.forEach(pair ->
+                {
+                    switch (pair.key().kind())
+                    {
+                    case 1:
+                        AmqpULongFW id = amqpULongRW.wrap(valueBuffer, valueOffset.value, valueBuffer.capacity())
+                            .set(pair.key().id()).build();
+                        valueOffset.value += id.limit();
+                        AmqpByteFW value1 = amqpByteRW.wrap(valueBuffer, valueOffset.value, valueBuffer.capacity())
+                            .set(pair.value().bytes().value().getByte(0)).build();
+                        valueOffset.value += value1.limit();
+                        amqpMapRW.entry(k -> k.setAsAmqpULong(id), v -> v.setAsAmqpByte(value1));
+                        break;
+                    case 2:
+                        AmqpSymbolFW name = amqpSymbolRW.wrap(valueBuffer, valueOffset.value, valueBuffer.capacity())
+                            .set(pair.key().name()).build();
+                        valueOffset.value += name.limit();
+                        AmqpByteFW value2 = amqpByteRW.wrap(valueBuffer, valueOffset.value, valueBuffer.capacity())
+                            .set(pair.value().bytes().value().getByte(0)).build();
+                        valueOffset.value += value2.limit();
+                        amqpMapRW.entry(k -> k.setAsAmqpSymbol(name), v -> v.setAsAmqpByte(value2));
+                        break;
+                    }
+                });
+                messageAnnotations = amqpMapRW.build();
+            }
+            keyOffset.value = 0;
+            valueOffset.value = 0;
+            return messageAnnotations;
+        }
+
+        private AmqpPropertiesFW properties(
+            ArrayFW<AmqpMessagePropertyFW> propertiesArray,
+            int bufferOffset)
+        {
+            AmqpPropertiesFW properties = null;
+            if (propertiesArray.fieldCount() > 0)
+            {
+                amqpPropertiesRW.wrap(extraBuffer, bufferOffset, extraBuffer.capacity());
+                propertiesArray.forEach(field ->
+                {
+                    switch (field.kind())
+                    {
+                    case 0:
```

Comment:
Use constants from generated flyweight for `kind` values to make the code more readable.